### PR TITLE
auth-4.1.x: set up circleCI testing for auth+godbc+mssql

### PR DESCRIPTION
**Pull request by Habbie**
(cherry picked from commit fbd89850013536ecad8da2368d136f092d355388)

### Short description
Backport CircleCi ODBC testing to auth 4.1.x.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Comment by aerique** [comment]
Why does this PR not show CircleCI checks?

**Comment by Habbie** [comment]
> Why does this PR not show CircleCI checks?

Excellent question! I don't know :(

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          name: install dependencies
+          command: ./build-scripts/circleci.sh debian-stretch-deps
+
+      - run:
+          name: autoconf
+          command: autoreconf -vfi
+
+      - run:
+          name: configure
+          command: ./configure --disable-lua-records --with-modules='bind gmysql godbc random'
```
Comment (rgacogne) [inline .circleci/config.yml]: Out of curiosity, why do we build with `gmysql` and `random` here? I would have expected `gsqlite3` instead.

Comment (Habbie) [reply .circleci/config.yml]: No good reason at all, just what I ended up accidentally with after experimenting for a while.

**Pull request merged**